src/symmetry.jl

```diff
@@ -239,3 +239,81 @@ function check_symmetric(basis, ρin; tol=1e-10, symmetries=ρin.basis.model.sym
         @assert norm(symmetrize(ρin, [symop]) - ρin) < tol
     end
 end
+
+""""
+Convert a `basis` into one that doesn't use BZ symmetry.
+This is mainly useful for debug purposes (e.g. in cases we don't want to
+bother thinking about symmetries).
+"""
+function unfold_BZ(basis::PlaneWaveBasis)
+    if all(length.(basis.ksymops_global) .== 1)
+        return basis
+    else
+        kcoords = []
+        for (ik, kpt) in enumerate(basis.kcoords_global)
+            for (S, τ) in basis.ksymops_global[ik]
+                push!(kcoords, normalize_kpoint_coordinate(S * kpt))
+            end
+        end
+        new_basis = PlaneWaveBasis(basis.model,
+                                   basis.Ecut, basis.fft_size, basis.variational,
+                                   kcoords, [[identity_symop()] for _ in 1:length(kcoords)],
+                                   basis.kgrid, basis.kshift, basis.symmetries, basis.comm_kpts)
+    end
+end
+
+# find where in an irreducible basis kpoint `kcoord_unfolded` is handled
+function unfold_mapping(basis_irred, kpt_unfolded)
+    for ik_irred = 1:length(basis_irred.kpoints)
+        kpt_irred = basis_irred.kpoints[ik_irred]
+        for symop in basis_irred.ksymops[ik_irred]
+            if normalize_kpoint_coordinate(symop[1] * kpt_irred.coordinate) ≈
+                normalize_kpoint_coordinate(kpt_unfolded.coordinate) && kpt_unfolded.spin == kpt_irred.spin
+                return ik_irred, symop
+            end
+        end
+    end
+    error("Invalid unfolding of BZ")
+end
+
+function unfold_array(basis_irred, basis_unfolded, data, is_ψ)
+    if basis_irred == basis_unfolded
+        return data
+    end
+    if !(basis_irred.comm_kpts == basis_irred.comm_kpts == MPI.COMM_WORLD)
+        error("Brillouin zone symmetry unfolding not supported with MPI yet")
+    end
+    data_unfolded = similar(data, length(basis_unfolded.kpoints))
+    for ik_unfolded in 1:length(basis_unfolded.kpoints)
+        kpt_unfolded = basis_unfolded.kpoints[ik_unfolded]
+        ik_irred, symop = unfold_mapping(basis_irred, kpt_unfolded)
+        if !is_ψ
```

Comment:
How about better calling this flag apply_symops or something like that.

Reply:
I don't want to give the impression this is a general thing : it's really specialized for psi